expression/integration_test.go

```diff
@@ -2776,6 +2777,9 @@ func TestTiDBDecodeKeyFunc(t *testing.T) {
 	store, clean := testkit.CreateMockStore(t)
 	defer clean()
 
+	collate.SetNewCollationEnabledForTest(false)
+	defer collate.SetNewCollationEnabledForTest(true)
```

Comment:
Why did the test fail if new collation is enabled?

Reply:
EncodeKey depends on the collation, the previous result is based on the old collation